### PR TITLE
`solvers`: ParaSwap CLI

### PR DESCRIPTION
**Pull request by cowfee**
Follow-up to https://github.com/cowprotocol/services/pull/1513.

Fixes https://github.com/cowprotocol/services/issues/882.

Adds CLI and tests for the ParaSwap integration in `solvers`.

### Test Plan

Automated tests.

### CODE REVIEW

File: crates/solvers/src/lib.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 // TODO remove this once the crate stabilizes a bit.
 #![allow(dead_code)]
+#![recursion_limit = "256"]
```
Comment (vkgnosis) [inline crates/solvers/src/lib.rs:3]: Why?

**Review by vkgnosis — APPROVED** [review]

File: crates/solvers/src/lib.rs (hunk 1 above)
Comment (nlordell) [reply crates/solvers/src/lib.rs:3]: Also curious about this one.

Comment (cowfee) [reply crates/solvers/src/lib.rs:3]: Was required by some `json!` macro invocation for some reason. :shrug:

File: crates/solvers/src/domain/order.rs [hunk 2]
```diff
@@ -201,4 +202,5 @@ impl EcdsaSignature {
 /// This is a hash allowing arbitrary user data to be associated with an order.
 /// While this type holds the hash, the data itself is uploaded to IPFS. This
 /// hash is signed along with the order.
+#[derive(Clone, Copy, Debug, Default)]
```
[the remark this answers is not in the document]
Comment (cowfee) [reply crates/solvers/src/domain/order.rs]: https://github.com/cowprotocol/services/pull/1529/commits/7d9a98e0f9e0242f966d552bf61768f39fcecb15

**Pull request merged**